test/jellyfish_web/controllers/room_controller_test.exs

```diff
@@ -144,6 +144,25 @@ defmodule JellyfishWeb.RoomControllerTest do
       assert Enum.any?(rooms, &(&1.id == room2_id))
       assert Enum.all?(rooms, &(&1.id != room_id))
     end
+
+    test "roomService removes room on engine crash", %{room_id: room_id} = state do
+      %{room_id: room2_id} = create_room(state)
+
+      room_pid = RoomService.find_room!(room_id)
+
+      :erlang.trace(Process.whereis(RoomService), true, [:receive])
+
+      %{engine_pid: engine_pid} = :sys.get_state(room_pid)
+
+      assert true = Process.exit(engine_pid, :error)
+
+      assert_receive({:trace, _pid, :receive, {:DOWN, _ref, :process, ^room_pid, :error}})
```

Comment:
We don't need this and `:erlang.trace` too? It should be enought to make sure `list_rooms` doesnt return room with the id `room_id`

Reply:
Nope, because without it, `list_roms` returns room with id `room_id`. I think that because a room isn't removed from the registry fast enough and from that we get race condition.